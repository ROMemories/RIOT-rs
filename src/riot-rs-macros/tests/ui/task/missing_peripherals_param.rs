#![no_main]
#![feature(type_alias_impl_trait)]
#![feature(used_with_arg)]

// FAIL: the `peripherals` parameter is required in this case
#[riot_rs::task(autostart)]
async fn main(_peripherals: Peripherals) {}

struct Peripherals;
