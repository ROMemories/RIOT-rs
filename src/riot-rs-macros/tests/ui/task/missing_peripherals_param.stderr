error: custom attribute panicked
 --> tests/ui/task/missing_peripherals_param.rs:7:1
  |
7 | #[riot_rs::task(autostart)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: to provide this function with peripherals, use the `peripherals` macro parameter
